src/main/java/org/folio/linked/data/integration/kafka/listener/handler/SourceRecordDomainEventHandler.java

```diff
@@ -0,0 +1,118 @@
+    + "thus skipping it's update out of SourceRecordDomainEvent [id {}]";
+  private static final Set<SourceRecordType> SUPPORTED_RECORD_TYPES = Set.of(MARC_BIB, MARC_AUTHORITY);
+  private static final Set<SourceRecordDomainEvent.EventTypeEnum> SUPPORTED_EVENT_TYPES = Set.of(CREATED, UPDATED);
+  private final MarcBib2ldMapper marcBib2ldMapper;
+  private final MarcAuthority2ldMapper marcAuthority2ldMapper;
+  private final ResourceService resourceService;
+  private final InstanceMetadataRepository instanceMetadataRepository;
+
+  public void handle(SourceRecordDomainEvent event, SourceRecordType recordType) {
+    if (notValidEventData(event, recordType)) {
+      return;
+    }
+    if (recordType == MARC_BIB) {
+      saveBib(event);
+    } else if (recordType == MARC_AUTHORITY) {
+      saveAuthorities(event);
+    }
+  }
+
+  private boolean notValidEventData(SourceRecordDomainEvent event, SourceRecordType recordType) {
+    if (isEmpty(event.getEventPayload())) {
+      log.warn(NO_MARC_EVENT, event.getId());
+      return true;
+    }
+    if (isNull(recordType) || !SUPPORTED_RECORD_TYPES.contains(recordType)) {
+      logUnsupportedType(event, "record", recordType);
+      return true;
+    }
+    if (isNull(event.getEventType()) || !SUPPORTED_EVENT_TYPES.contains(event.getEventType())) {
+      logUnsupportedType(event, "event", event.getEventType());
+      return true;
+    }
+    return false;
+  }
+
+  private void saveBib(SourceRecordDomainEvent event) {
+    marcBib2ldMapper.fromMarcJson(event.getEventPayload())
+      .ifPresentOrElse(resource -> {
+          if (isInstanceWithLinkedDataSource(resource)) {
+            log.info(IGNORED_LINKED_DATA_SOURCE, resource.getId(), event.getId());
+          } else {
+            saveResource(resource, event);
+          }
+        },
+        () -> logEmptyResource(event.getId())
+      );
+  }
+
+  private boolean isInstanceWithLinkedDataSource(Resource resource) {
+    return resource.getTypes().equals(Set.of(INSTANCE))
+      && instanceMetadataRepository.findById(resource.getId())
+      .map(InstanceMetadata::getSource)
+      .map(LINKED_DATA::equals)
+      .orElse(false);
+  }
+
+  private void saveAuthorities(SourceRecordDomainEvent event) {
+    var mapped = marcAuthority2ldMapper.fromMarcJson(event.getEventPayload());
+    if (mapped.isEmpty()) {
+      logEmptyResource(event.getId());
+    } else {
+      mapped.forEach(resource -> saveResource(resource, event));
+    }
+  }
+
+  private void saveResource(Resource resource, SourceRecordDomainEvent event) {
+    if (CREATED == event.getEventType() || UPDATED == event.getEventType()) {
+      var id = resourceService.saveMarcResource(resource);
+      log.info(EVENT_SAVED, event.getId(), id);
+    }
+  }
+
+  private void logUnsupportedType(SourceRecordDomainEvent event, String typeKind, Enum<?> type) {
+    log.warn(UNSUPPORTED_TYPE, typeKind, isNull(type) ? "null" : type.name(), event.getId());
```

Comment:
May be we have to do debug log here? I assume SRS will be emitting events of types MARC_HOLDING & EDIFACT. If we log a WARN message for each such event, it will give a wrong impression that something is wrong with the application.

Reply:
Agree, let's use info level. Done